tests/test_intake_xarray.py

```diff
@@ -138,3 +139,40 @@ def test_read_partition_tiff():
     d = s.to_dask().data
     expected = d[:1].compute()
     assert np.all(out == expected)
+
+
+def test_read_pattern_concat_on_existing_dim():
+    pytest.importorskip('rasterio')
+    cat = intake.open_catalog(os.path.join(here, 'data', 'catalog.yaml'))
+    colors = cat.pattern_tiff_source_concat_on_band()
+
+    da = colors.read()
+    assert len(da.color) == 6
```

Comment:
Here you just check the entries for the given coordinate, but it would be good to check that the output array also has the same axis correctly set.